azure-pipelines-dependency-graph-testing.yml

```diff
@@ -0,0 +1,503 @@
+# This pipeline is meant to run the GE test suite with an experimental test runner strategy.
+# The significant changes between this YAML and the primary azure-pipelines.yml file is
+# that this only tests Python 3.8 (for performance considerations) and utilizes a
+# custom script to filter the test files selected and passed on to pytest.
+trigger:
+  branches:
+    include:
+    - pre_pr-*
+    - develop
+    - main
```

Comment:
Let me know if this should be streamlined further (or conversely, more closely follow the main YAML).

An alternative approach would be to add this change in our current pipeline but as an additional stage in our compatability or comprehensive matrices. This would introduce the change to production but reduce the overall burden of the experiment.